osnap/analytics/dynamics.py

```diff
@@ -7,19 +7,22 @@
 import scipy.spatial.distance as d
 
 import libpysal
+from libpysal.weights.contiguity import Rook, Queen
+from libpysal.weights.distance import KNN, Kernel
 
 
 class Transition(object):
     """
     (Spatial) Markov approaches to transitional dynamics of neighborhood types.
 
     Parameters
-    ----------
-    y               : array
-                      (n,t), one row per observation, one column per neighborhood
-                      type of each spatial unit, with as many columns as time periods.
-    w               : W
+    ----------  
+    dataset          : osnap.Dataset
```

Comment:
I am fine with doing it this way. But I think this will result in code duplication on dealing with input data in `Transition` and `Sequence` classes or other longitudinal methods to be implemented in the future. 

I am thinking that we can have a more general warper class sitting on top of `Transition` and `Sequence` classes to deal with the input datasets (either pandas's Dataframe and a weight file, or geopandas's GeoDataFrame and weight type, or numpy array and a weight file, etc) so that we can avoid duplicate code in dealing with input data for these two (or more) longitudinal analysis methods. Another way to handle this would be to make `Transition` and `Sequence` classes to inherit a base class which has methods to deal with different input datasets.

Reply:
that sounds great. i'd really intended for this to be in a separate branch but it got caught up in the merge conflicts my accident